TrackingSystem/src/main/java/de/telran/controller/CustomerController.java

```diff
@@ -1,33 +1,79 @@
 package de.telran.controller;
 
+import de.telran.dto.CustomerDTO;
 import de.telran.entity.Customer;
+import de.telran.entity.Shipment;
 import de.telran.service.CustomerService;
+import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 @RestController
 public class CustomerController {
 
     CustomerService customerService;
 
+    ModelMapper modelMapper = new ModelMapper();
+
     @Autowired
     public CustomerController(CustomerService customerService) {
         this.customerService = customerService;
     }
 
     @GetMapping("/api/customers")
-    List<Customer> getAllCustomers() {
-        return customerService.getAllCustomers();
+    List<CustomerDTO> getAllCustomers() {
+        return  customerService.getAllCustomers()
+                .stream()
+                .map(this::convertCustomerToCustomerDTO)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("api/customers/{customer_id}/shipments")
+    CustomerDTO getAllShipmentsOfACustomer(@PathVariable("customer_id") Long id) {
```

Comment:
This is confusing: your URL path tells us: I return a collection of shipments for a customer. 
But your method returns a CustomerDTO, which means it returns a single object, not a collection, as specified by the URL path. 

I think in this are there are 2 options:
1. You return a list of shipments of a customer with specified ID, which customer_id of every shipment is returned as a Long value
2. You return a list of shipments for a customer, where every shipment has a field Customer with full json of a customer. (we discussed this many to one association last time). 

Which solution to use depends on your requirements and front end plans, but the confusion I mentioned is better to be eliminated.